File: targets/js/sample-counter-v3/lib/views/counter-view.js
export class CounterView {
    _text = null;
    onButtonClick = null;
    constructor() { }
    render(container) {
        const root = document.createElement("div");
        container.append(root);
        const text = document.createElement("span");
        text.innerHTML = "0";
        root.append(text);
        this._text = text;
        const button = document.createElement("button");
        button.innerHTML = "Click me";
        button.onclick = (_) => this.onButtonClick?.();
        root.append(button);
    }
    showCounter(counter) {
        this._text != null && (this._text.innerHTML = counter.toString());
    }
}

File: samples/SampleCounterV1/Views/CounterView.cs
using Metano.TypeScript.DOM;
using SampleCounterV1.Presenters;

namespace SampleCounterV1.Views;

public sealed class CounterView : ICounterView
{
    private HtmlSpanElement? _text;

    public Action? OnButtonClick { get; set; }

    public void Render(HtmlElement container)
    {
        var root = Js.Document.CreateElement(HtmlElementType.Div);
        container.Append(root);

        var text = Js.Document.CreateElement(HtmlElementType.Span);
        text.InnerHtml = "0";
        root.Append(text);
        _text = text;

        var button = Js.Document.CreateElement(HtmlElementType.Button);
        button.InnerHtml = "Click me";
        button.OnClick = _ => OnButtonClick?.Invoke();
        root.Append(button);
    }

    public void ShowCounter(int counter)
    {
        _text?.InnerHtml = counter.ToString();
    }
}
